src/pages/index.js

```diff
@@ -11,86 +11,198 @@ import {
+    cardSection = new Section(
+      {
+        items: cards,
+        renderer: renderCard,
+      },
+      cardSettings.cardList
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+function createCard(data) {
   const cardElement = new Card(
     {
-      data: item,
-      handleImageClick: () => {
-        cardImagePreview.open(item);
+      data,
+      handleImageClick: (imgData) => {
+        cardImagePreview.open(imgData);
+      },
+      handleDelete: () => {
+        confirmPreviewPopup.open();
+        confirmPreviewPopup.submitBtnAction(() => {
+          confirmPreviewPopup.renderLoading(true);
+          api
+            .deleteCard(data._id)
+            .then((res) => {
+              cardElement.handleDeleteButton(res);
+              confirmPreviewPopup.close();
+            })
+            .catch((err) => console.log(`An error occured: ${err}`))
+            .finally(() => confirmPreviewPopup.renderLoading(false));
+        });
       },
+      handleLike: () => {
+        if (!cardElement.isCardLiked()) {
+          return api
+            .likeCard(cardElement._cardId)
+            .then((res) => {
+              cardElement.setlikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.error(err);
+            });
+        } else {
+          return api
+            .disLikeCard(cardElement._cardId)
+            .then((res) => {
+              cardElement.setlikes(res.isLiked);
+            })
+            .catch((err) => {
+              console.error(err);
+            });
+        }
+      },
+      userId: userInfo.getId(),
     },
     cardSettings.cardTemplate
   );
   return cardElement.getView();
 }
 
-const cardLayout = new Section(
-  {
-    items: initialCards,
-    renderer: (data) => {
-      const cardEl = createCard(data);
-
-      cardLayout.addItem(cardEl);
-    },
-  },
-  cardSettings.cardList
-);
-cardLayout.renderItems();
+const renderCard = (data) => {
+  const cardEl = createCard(data);
+  cardSection.addItem(cardEl);
+};
 
 const userInfoPopup = new PopupWithForm({
   popupSelector: profileEditModal,
   handleFormSubmit: (userData) => {
-    userInfo.setUserInfo(userData);
-    userInfoPopup.close();
+    userInfoPopup.renderLoading(true);
+    api
+      .editProfileApi(userData)
+      .then((userData) => {
+        userInfo.setUserInfo(userData);
+        userInfoPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => userInfoPopup.renderLoading(false));
   },
 });
 
 const newCardPopup = new PopupWithForm({
   popupSelector: addCardModal,
   handleFormSubmit: (data) => {
-    const card = createCard(data);
-    cardLayout.addItem(card);
-    newCardPopup.close();
+    newCardPopup.renderLoading(true);
+    api
+      .createCardApi(data)
+      .then((data) => {
+        renderCard(data);
+        newCardPopup.close();
+        addFormValidator.disableButton();
```

Comment:
Needs correction: You should disable the button only before opening the popup, since you reset the form every time you close the popup